.github/workflows/test.yml

```diff
@@ -80,6 +80,11 @@ jobs:
       run: |
         pip install git+https://github.com/NeurodataWithoutBorders/pynwb
 
+    - name: Install dev version of nwbinspector
+      if: matrix.mode == 'dev-deps'
+      run: |
+        pip install git+https://github.com/NeurodataWithoutBorders/nwbinspector
```

Comment:
Went ahead and added the NWB Inspector dev branch to the `dev-deps` testing mode alongside PyNWB.

Let me know if you'd prefer me to revert this, though.

Reply:
Now, what is a bit odd here, when I ran the testing suite on a local clone of `dandi-cli` with the dev branch of `nwbinspector` also installed in the same virtualenv, all the tests passed.

Reason I find that odd, is the fixtures define metadata that ought to be flagged if those fixtures are being used to test dandi validation, specifically this line: https://github.com/dandi/dandi-cli/blob/master/dandi/tests/fixtures.py#L117, which should be "mus musculus" to pass the NWB Inspector.

I could not tell at a glance, but where in the testing suite are those fixtures used to test validation of NWB assets?

Reply:
I would fold this `pip install` into previous step and just adjust description that it is dev version of both pynwb and nwbinspector.

Rationale
- do not duplicate conditioning on dev-deps
- shorter

Reply:
> I could not tell at a glance, but where in the testing suite are those fixtures used to test validation of NWB assets?

```
$> git grep simple2_nwb
dandi/cli/tests/test_command.py:def test_smoke(simple2_nwb, command):
dandi/cli/tests/test_command.py:    r = runner.invoke(command, [simple2_nwb])
dandi/tests/fixtures.py:def simple2_nwb(
dandi/tests/fixtures.py:    simple2_nwb: str, tmp_path_factory: pytest.TempPathFactory
dandi/tests/fixtures.py:        organize, ["-f", "copy", "--dandiset-path", str(tmp_path), str(simple2_nwb)]
dandi/tests/fixtures.py:    simple2_nwb: str,
dandi/tests/fixtures.py:    shutil.copy(simple2_nwb, tmp_path)
dandi/tests/test_files.py:def test_validate_simple2(simple2_nwb):
dandi/tests/test_files.py:    errors = dandi_file(simple2_nwb).get_validation_errors()
dandi/tests/test_files.py:def test_validate_simple2_new(simple2_nwb):
dandi/tests/test_files.py:    errors = dandi_file(simple2_nwb).get_validation_errors(
dandi/tests/test_organize.py:def test_ambiguous(simple2_nwb: str, tmp_path: Path) -> None:
dandi/tests/test_organize.py:    copy2 = copy_nwb_file(simple2_nwb, tmp_path)
dandi/tests/test_organize.py:    args = ["--files-mode", "copy", "-d", outdir, simple2_nwb, copy2]
dandi/tests/test_organize.py:        for f in [simple2_nwb, copy2]
```
and since used in those other fixtures:

```shell
$> git grep organized_nwb_dir | grep test_ 
dandi/tests/test_fixtures.py:def test_organized_nwb_dir(organized_nwb_dir: Path) -> None:
dandi/tests/test_fixtures.py:def test_organized_nwb_dir2(organized_nwb_dir2: Path) -> None:
dandi/tests/test_upload.py:    new_dandiset: SampleDandiset, organized_nwb_dir: str, tmp_path: Path
dandi/tests/test_upload.py:        p for p in list_paths(organized_nwb_dir) if p.name != dandiset_metadata_file
dandi/tests/test_upload.py:    parent, name = nwb_file.relative_to(organized_nwb_dir).parts
```

Reply:
> I would fold this pip install into previous step and just adjust description that it is dev version of both pynwb and nwbinspector.
> 
> Rationale
> 
> do not duplicate conditioning on dev-deps
> shorter

Sure thing

Reply:
> and since used in those other fixtures:

Ah, thanks so much! I've updated the fixtures to the new requirements as well.